### PR TITLE
Multiredis

### PR DESCRIPTION
**Pull request by iamigo**
Allows admin to pin each of the various redis use cases to a specific redis instance based on config.

Note: Leaving out docs for now, pending the creation of a doc on moving Refocus to production (separate user story).

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -114,6 +111,25 @@ module.exports = {
...
+        pe.REDIS_PUBSUB || pe.REDIS_URL || DEFAULT_LOCAL_REDIS_URL,
+      queue:
+        pe.REDIS_QUEUE || pe.REDIS_URL || DEFAULT_LOCAL_REDIS_URL,
+      realtimeLogging:
+        pe.REDIS_REALTIME_LOGGING || pe.REDIS_URL || DEFAULT_LOCAL_REDIS_URL,
+      sampleStore:
+        pe.REDIS_SAMPLE_STORE || pe.REDIS_URL || DEFAULT_LOCAL_REDIS_URL,
+      session:
+        pe.REDIS_SESSION || pe.REDIS_URL || DEFAULT_LOCAL_REDIS_URL,
+    },
```
Comment (shriramshankar) [inline config.js]: do we need a different redis for session, realtimeLogging and limiter?

Comment (iamigo) [inline config.js]: We don't *need* to... but once I was identifying all the different uses, I just went with making *all* of them configurable, giving admin the ability to leave stuff on the main redis instance or move anything off to a separate instance.

Comment (shriramshankar) [inline config.js]: Also, if an admin configures a separate redis instance for each of the above use cases nothing will remain on the main redis.  Are we ok with this? It does not matter to me, though.

File: cache/redisCache.js [hunk 2]
```diff
@@ -7,19 +7,23 @@
...
 
 module.exports = {
-  client,
+  client: {
+    cache: redis.createClient(rconf.instanceUrl.cache),
+    limiter: redis.createClient(rconf.instanceUrl.limiter),
+    pub: redis.createClient(rconf.instanceUrl.pubsub),
+    realtimeLogging: redis.createClient(rconf.instanceUrl.realtimeLogging),
+    sampleStore: redis.createClient(rconf.instanceUrl.sampleStore),
+    sub,
```
Comment (shriramshankar) [inline cache/redisCache.js]: do we want to export the session redis client too?

Comment (iamigo) [inline cache/redisCache.js]: No, the redis session store constructor take a redis url, not a redis client

File: config.js (hunk 1 above)
Comment (iamigo) [inline config.js]: I'm ok with giving choice to admin.

File: config.js [hunk 3]
```diff
@@ -74,8 +75,28 @@ const deprioritizeJobsFrom = configUtil.csvToArray(pe.DEPRIORITIZE_JOBS_FROM);
...
 const JOB_QUEUE_TTL_SECONDS = pe.TTL_KUE_JOBS || DEFAULT_JOB_QUEUE_TTL_SECONDS;
 
-module.exports = {
+/*
+ * Assigns each of the different redis uses cases to a particular redis
+ * instance, if configured, or falls back to the primary redis instance.
+ */
+const redisUrls = {
+  cache: pe.REDIS_CACHE && pe[pe.REDIS_CACHE] ?
+    pe[pe.REDIS_CACHE] : (pe.REDIS_URL || DEFAULT_LOCAL_REDIS_URL),
```
Comment (pallavi2209) [inline config.js]: Can you give an example here, a bit hard to interpret.
So if we have pe.REDIS_URL (lets say x) set, the result will be x, else the result will be local redis url. But if pe.REDIS_CACHE is set (lets say y), do you expect the result to be y or x?

Comment (pallavi2209) [inline config.js]: Okay, got it, && has preference over ?
And to get configured cache working we need to have pe.REDIS_CACHE set (say x) and pe[x] set (say y)

**Pull request merged**